### PR TITLE
Render non-home token reservations

### PR DESCRIPTION
**Pull request by michaeljb**
Also:
- add Erie's token ability
- fix a bug with the static 1846 map page where Erie's home token is rendered

Screenshot from 2020-06-22 21-29-02

Screenshot from 2020-06-22 21-24-46

### CODE REVIEW

File: assets/app/view/map_page.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def render
 
       return h(:p, "Bad game title: #{game_title}") unless game_class
 
-      names = %w[p1 p2 p3 p4]
+      names = %w[p1 p2 p3 p4 p5]
```
Comment (tobymao) [inline assets/app/view/map_page.rb:17]: this breaks games that don't support 5 players like 1836jr.

in lib/engine.rb i have a function called player_range

File: lib/engine/game/base.rb [hunk 2]
```diff
@@ -560,6 +559,12 @@ def init_hexes(companies, corporations)
           end
         end
 
+        reservations = Hash.new { |k, v| k[v] = [] }
+        corporations.each do |corp|
+          reservations[corp.coordinates].unshift(corp)
```
Comment (tobymao) [inline lib/engine/game/base.rb:564]: why are we unshifting here?

Comment (michaeljb) [reply lib/engine/game/base.rb:564]: It ensures home tokens are always first in the reservation list for cities like Erie in 1846 with reservations for multiple corps. Not essential though.

**Comment by michaeljb** [comment]
Going to re-do this as a new reservation ability.

**Pull request closed without merging**